File: vdp-java/src/test/java/com/visa/vdp/vctc/TestProgramAdministration.java
package com.visa.vdp.vctc;

import java.util.HashMap;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.apache.http.HttpStatus;
import org.testng.annotations.Test;

import com.visa.vdp.utils.AbstractVisaAPIClient;
import com.visa.vdp.utils.MethodTypes;

public class TestProgramAdministration {
	
	AbstractVisaAPIClient abstractVisaAPIClient;
	
	@Test(groups = "vctc")
    public void setUp() {
   	    this.abstractVisaAPIClient = new AbstractVisaAPIClient();
	}
	
	@Test(groups = "vctc")
	public void testRetreiveTransactionTypeControls() throws Exception {
	    String baseUri = "vctc/";
	    String resourcePath = "programadmin/v1/configuration/transactiontypecontrols";
	    
	    CloseableHttpResponse response = this.abstractVisaAPIClient.doMutualAuthRequest(baseUri + resourcePath, "Retrieve Transaction Type Controls", "", MethodTypes.GET, new HashMap<String, String>());
	    Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
	    response.close();
	}

}

File: vdp-python/visa/test/vctc/test_program_administration.py
from visa.helpers.abstract_visa_api_client import AbstractVisaAPIClient
import unittest
'''
@author: visa
'''

class TestProgramAdministration(unittest.TestCase):
    
    def setUp(self):
        self.abstract_visa_api_client = AbstractVisaAPIClient()
    
    def test_retrieve_transaction(self):
        base_uri = 'vctc/'
        resource_path = 'programadmin/v1/configuration/transactiontypecontrols'
        response = self.abstract_visa_api_client.do_mutual_auth_request(base_uri + resource_path, '', 'Retrieve Transaction Type Control call', 'get')
        self.assertEqual(str(response.status_code) ,"200" ,"Retrieve transaction type control test failed")
        pass